pkg/kubelet/kubelet.go

```diff
@@ -110,8 +110,8 @@ const (
 	// Max amount of time to wait for the container runtime to come up.
 	maxWaitForContainerRuntime = 30 * time.Second
 
-	// nodeStatusUpdateRetry specifies how many times kubelet retries when posting node status failed.
-	nodeStatusUpdateRetry = 5
+	// nodeStatusUpdateRetryBackoff is the amount of time that we are willing to spend retrying to connect to the apiserver.
+	nodeStatusUpdateRetryMaximumTime = 20 * time.Minute
```

Comment:
> you may have a pod that doesn't need network at all to run (e.g. some batch job). And we don't want to evict it.

I guess its a debate for another issue, but Batch jobs can be very tricky to coordinate in a large distributed system and often can be broken or bring a machine down, and need to be killed i.e. these get hit all the time in the bigdata space https://en.wikipedia.org/wiki/Spider_trap , and misconfigured stream or batch frameworks will come to a halt and maybe even bring down a machine if they hit them.  Consider especially the tendency for the JVM (which often is the basis for those batch jobs) to take over a cpu and memory, greedily use off heap resources, and so on.